sdk/nodejs/lambda/metrics.ts

```diff
@@ -139,3 +141,97 @@ export module metrics {
+             *
+             * * Permissions errors
+             * * Throttles from downstream services
+             * * Misconfigured resources
+             * * Timeouts
+             */
+            deadLetterErrors(change?: cloudwatch.MetricChange): cloudwatch.Metric;
+
+            /**
+             * Measures the elapsed wall clock time from when this function code starts executing as a
+             * result of an invocation to when it stops executing. The maximum data point value possible
+             * is the function timeout configuration. The billed duration will be rounded up to the
+             * nearest 100 millisecond. Note that AWS Lambda only sends these metrics to CloudWatch if
+             * they have a nonzero value.
+             */
+            duration(change?: cloudwatch.MetricChange): cloudwatch.Metric;
+
+            /**
+             * Measures the number of this Lambda function's invocation attempts that were throttled due
+             * to invocation rates exceeding the customer’s concurrent limits (error code 429). Failed
+             * invocations may trigger a retry attempt that succeeds.
+             */
+            throttles(change?: cloudwatch.MetricChange): cloudwatch.Metric;
+
+            /**
+             * Emitted for stream-based invocations only (functions triggered by an Amazon DynamoDB
+             * stream or Kinesis stream). Measures the age of the last record for each batch of records
+             * processed. Age is the difference between the time this Lambda received the batch, and the
+             * time the last record in the batch was written to the stream.
+             */
+            iteratorAge(change?: cloudwatch.MetricChange): cloudwatch.Metric;
+        }
+    }
+}
+
+// All instance metrics just make a normal AWS/Lambda metric, except with the FunctionName
+// dimension set to this Function's name.
+
+function getMetric(
+    func: LambdaFunction, change: cloudwatch.MetricChange,
+    moduleFunction: (change: cloudwatch.MetricChange) => cloudwatch.Metric) {
+
+    return moduleFunction({ dimensions: { FunctionName: func.name } }).with(change);
+}
+
+Object.defineProperty(LambdaFunction.prototype, "metrics", {
+    get: function (this: LambdaFunction) {
+        const lambda = this;
+        return {
+            invocations: (change: cloudwatch.MetricChange) => getMetric(lambda, change, metrics.invocations),
+            errors: (change: cloudwatch.MetricChange) => getMetric(lambda, change, metrics.errors),
+            deadLetterErrors: (change: cloudwatch.MetricChange) => getMetric(this, change, metrics.deadLetterErrors),
+            duration: (change: cloudwatch.MetricChange) => getMetric(lambda, change, metrics.duration),
+            throttles: (change: cloudwatch.MetricChange) => getMetric(lambda, change, metrics.throttles),
+            iteratorAge: (change: cloudwatch.MetricChange) => getMetric(lambda, change, metrics.iteratorAge),
+        }
+    }
+});

```

Comment:
need to define hte 'metrics' property this way as opposed to just attaching something to the prototype.  This is necessary so we can have 'metrics' be a 'getter' function, allowing us access to the right 'this' value.  We need to stash that away so we can pass it along to all the sub-functions properly.